### DISCUSSION TITLE
Contributing and reviewing guidelines

### DISCUSSION THREAD

**Discussion by StevenClontz** [pi-base/data#322] [Ideas]
https://github.com/pi-base/data/wiki/Contributing

Take a look at the wiki above - I've updated it a bit based on my understanding from yesterday's conversation. Please feel free to discuss these in this thread, suggest improvements, etc.

**Comment by prabau** [comment]
Looks pretty good.

One thing we discussed after you were out of the call relates to this paragraph:
> Contributions that add verifyable mathematical knowledge to the pi-Base with appropriate references should be merged as soon as possible.
>   If the PR could nonetheless could be improved (e.g. a better choice of naming, a more useful description), those improvements should be proposed in a separate issue/PR (whether by encouraging the original contirbutor, or directly by the reviewer).

Making followup improvements in a separate PR makes sense for spaces and traits.  But for theorems we thought it may be better to apply higher scrutiny before merging.  In particular make sure that a theorem cannot be deduced from other ones and makes sense in the overall graph of implications.  Otherwise we run into the situation of accepting a theorem and immediately following that with a PR that removes the theorem or changes it beyong recognition, which seems counterproductive (even to the morale of the contributor).
